logging/src/main/java/org/jboss/as/logging/handlers/SocketHandlerResourceDefinition.java

```diff
@@ -0,0 +1,472 @@
+import org.jboss.as.logging.ElementAttributeMarshaller;
+import org.jboss.as.logging.KnownModelVersion;
+import org.jboss.as.logging.Logging;
+import org.jboss.as.logging.LoggingExtension;
+import org.jboss.as.logging.LoggingOperations;
+import org.jboss.as.logging.LoggingOperations.LoggingAddOperationStepHandler;
+import org.jboss.as.logging.TransformerResourceDefinition;
+import org.jboss.as.logging.capabilities.Capabilities;
+import org.jboss.as.logging.logging.LoggingLogger;
+import org.jboss.as.network.OutboundSocketBinding;
+import org.jboss.as.network.SocketBindingManager;
+import org.jboss.dmr.ModelNode;
+import org.jboss.dmr.ModelType;
+import org.jboss.logmanager.config.HandlerConfiguration;
+import org.jboss.logmanager.config.LogContextConfiguration;
+import org.jboss.logmanager.handlers.ClientSocketFactory;
+import org.jboss.logmanager.handlers.DelayedHandler;
+import org.jboss.logmanager.handlers.SocketHandler;
+import org.jboss.msc.Service;
+import org.jboss.msc.service.ServiceBuilder;
+import org.jboss.msc.service.ServiceName;
+import org.jboss.msc.service.StartContext;
+import org.jboss.msc.service.StopContext;
+import org.wildfly.common.function.Functions;
+
+/**
+ * Represents a {@link SocketHandler}. The handler will be wrapped with a {@link DelayedHandler} for booting from the
+ * {@code logging.properties} file.
+ *
+ * @author <a href="mailto:jperkins@redhat.com">James R. Perkins</a>
+ */
+@SuppressWarnings("Convert2Lambda")
+public class SocketHandlerResourceDefinition extends TransformerResourceDefinition {
+    public static final String NAME = "socket-handler";
+    private static final PathElement PATH = PathElement.pathElement(NAME);
+
+    public static final SimpleAttributeDefinition BLOCK_ON_RECONNECT = SimpleAttributeDefinitionBuilder.create("block-on-reconnect", ModelType.BOOLEAN, true)
+            .setAllowExpression(true)
+            .setDefaultValue(new ModelNode(false))
+            .build();
+
+    public static final SimpleAttributeDefinition NAMED_FORMATTER = SimpleAttributeDefinitionBuilder.create(AbstractHandlerDefinition.NAMED_FORMATTER)
+            .setAttributeMarshaller(ElementAttributeMarshaller.NAME_ATTRIBUTE_MARSHALLER)
+            .setAlternatives(new String[0])
+            .setRequired(true)
+            .build();
+
+    public static final SimpleAttributeDefinition OUTBOUND_SOCKET_BINDING_REF = SimpleAttributeDefinitionBuilder.create("outbound-socket-binding-ref", ModelType.STRING, false)
+            .addAccessConstraint(SensitiveTargetAccessConstraintDefinition.SOCKET_BINDING_REF)
+            .setAllowExpression(true)
+            .setCapabilityReference(Capabilities.OUTBOUND_SOCKET_BINDING_CAPABILITY)
+            .addFlag(AttributeAccess.Flag.RESTART_RESOURCE_SERVICES)
+            .build();
+
+    public static final SimpleAttributeDefinition PROTOCOL = SimpleAttributeDefinitionBuilder.create("protocol", ModelType.STRING, true)
```

Comment:
Just a minor note for reviewers. The management audit logging for the `syslog-handler` there is a child resource of `protocol` with the values; `tcp`. `tls` and `upd`. For the Elytron `syslog-audit-log` there is an attribute called `transport` with the values `TCP`, `SSL_TCP` and `UDP`.

I went with `protocol` for the attribute name as the `SocketHandler` in the log manager has an enum named `Protocol` with the values of `TCP`, `SSL_TCP` and `UDP`. However maybe `transport` would be better to keep the names consistent. I can change this if it's agreed it should be changed.